addons/mail/static/tests/discuss_app/discuss_tests.js

```diff
@@ -827,6 +827,63 @@ QUnit.test("rendering of inbox message", async (assert) => {
     assert.containsOnce($message, "[title='Mark as Read']");
 });
 
+QUnit.test('Unfollow message', async function (assert) {
+    assert.expect(11);
+
+    const pyEnv = await startServer();
+    const currentPartnerId = pyEnv.currentPartnerId;
+    const [threadFollowedId, threadNotFollowedId] = pyEnv["res.partner"].create([{
+        name: 'Thread followed',
+    }, {
+        name: 'Thread not followed',
+    }]);
+    const followerId = pyEnv["mail.followers"].create({
+        partner_id: currentPartnerId,
+        res_id: threadFollowedId,
+        res_model: "res.partner",
+    });
+    for (let [threadId, user_follower_id] of [
+        [threadFollowedId, followerId],
+        [threadFollowedId, followerId],
+        [threadNotFollowedId, false],
+    ]) {
+        const messageId = pyEnv["mail.message"].create({
+            body: "not empty",
+            model: "res.partner",
+            needaction: true,
+            needaction_partner_ids: [currentPartnerId],
+            res_id: threadId,
+            user_follower_id: user_follower_id,
+        });
+        pyEnv["mail.notification"].create({
+            mail_message_id: messageId,
+            notification_status: "sent",
+            notification_type: "inbox",
+            res_partner_id: currentPartnerId,
+        });
+    }
+    const { openDiscuss } = await start();
+    await openDiscuss();
+    // 2 messages about "Thread followed" with unfollow button and 1 message about "Thread not followed" without it
+    assert.containsN($, ".o-mail-Message", 3);
+    for (let messageN of [0, 1]) {
+        const $message = $(`.o-mail-Message:eq(${messageN})`);
+        assert.containsOnce($message, ".o-mail-Message-header:contains(on Thread followed)");
+        assert.containsOnce($message, "[title='Unfollow']");
+    }
+    const $messageNotFollowed = $(".o-mail-Message:eq(2)");
+    assert.containsOnce($messageNotFollowed, ".o-mail-Message-header:contains(on Thread not followed)");
+    assert.containsNone($messageNotFollowed, "[title='Unfollow']");
+
+    await click(".o-mail-Message:eq(0) button[title='Unfollow']");
+
+    assert.containsN($, ".o-mail-Message", 2, "Unfollowing message 0 marks it as read -> Message removed");
+    assert.containsN($, ".o-mail-Message-header:contains(on Thread followed)", 1);
+    assert.containsN($, ".o-mail-Message-header:contains(on Thread not followed)", 1);
```

Comment:
Note: `containsOnce` is typically used instead `containsN` with `1`

Reply:
Yes thanks. It’s updated.